GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/ClassificationPanelViewModel.cs

```diff
@@ -0,0 +1,174 @@
+﻿using GalaxyZooTouchTable.Lib;
+using GalaxyZooTouchTable.Models;
+using GalaxyZooTouchTable.Utility;
+using PanoptesNetClient;
+using PanoptesNetClient.Models;
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Windows.Input;
+
+namespace GalaxyZooTouchTable.ViewModels
+{
+    public class ClassificationPanelViewModel : INotifyPropertyChanged
+    {
+        public List<Subject> Subjects { get; set; } = new List<Subject>();
+        public Workflow Workflow { get; }
+        public WorkflowTask CurrentTask { get; set; }
+        public Classification CurrentClassification { get; set; }
+        public List<AnswerButton> CurrentAnswers { get; set; }
+        public Subject CurrentSubject { get; set; }
+        public string CurrentTaskIndex { get; set; }
+
+        private Annotation _currentAnnotation;
+        public Annotation CurrentAnnotation
+        {
+            get
+            {
+                return _currentAnnotation;
+            }
+            set
+            {
+                _currentAnnotation = value;
+                OnPropertyRaised("CurrentAnnotation");
+            }
+        }
+
+        private string _subjectImageSource;
+        public string SubjectImageSource
+        {
+            get
+            {
+                return _subjectImageSource;
+            }
+            set
+            {
+                _subjectImageSource = value;
+                OnPropertyRaised("SubjectImageSource");
+            }
+        }
+
+        public ICommand SelectAnswer { get; set; }
+        public ICommand SubmitClassification { get; set; }
+
+        private AnswerButton _selectedItem;
+        public AnswerButton SelectedItem
+        {
+            get
+            {
+                return _selectedItem;
+            }
+            set
+            {
+                _selectedItem = value;
+                OnPropertyRaised("SelectedItem");
+            }
+        }
+
+        public ClassificationPanelViewModel(Workflow workflow)
+        {
+            GetSubject();
+            LoadCommands();
+            Workflow = workflow;
+            CurrentTask = workflow.Tasks[workflow.FirstTask];
+            CurrentTaskIndex = workflow.FirstTask;
+
+            if (CurrentTask.Answers != null)
+            {
+                CurrentAnswers = ParseTaskAnswers(CurrentTask.Answers);
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyRaised(string propertyname)
+        {
+            PropertyChanged(this, new PropertyChangedEventArgs(propertyname));
+        }
+
+        private void LoadCommands()
+        {
+            SelectAnswer = new CustomCommand(ChooseAnswer, CanSelectAnswer);
+            SubmitClassification = new CustomCommand(SendClassification, CanSendClassification);
+        }
+
+        private async void SendClassification(object obj)
```

Comment:
For all of the following methods, can you use a more descriptive variable name than `obj` it's kind of unclear what that variable is for

Reply:
Of course. The `object obj` param is necessary to satisfy the `ICommand` interface, but changing it to `object sender` would help.